tests/push/test_ami_delete.py

```diff
@@ -0,0 +1,324 @@
+import os
+import re
+import pytest
+import json
+from mock import patch
+from pubtools._ami.tasks.delete import AmiDelete, entry_point
+
+AMI_PUB_SOURCE_ROOT = os.path.join(
+    os.path.dirname(__file__), "data/aws_pub_source/images.json"
+)
+
+
+@pytest.fixture()
+def mock_aws_delete():
+    with patch("pubtools._ami.services.aws.AWSService.delete") as m:
+        yield m
+
+
+@pytest.fixture()
+def images_json():
+    with open(AMI_PUB_SOURCE_ROOT) as f:
+        yield json.load(f)
+
+
+def _setup_test_ami_delete(
+    mock_aws_delete,
+    requests_mocker,
+    images_json,
+    rhsm_ami_ids,
+    keep_snapshot,
+    dry_run=False,
+):
+    mock_aws_delete.return_value = (
+        "ami-fake-id-01",
+        None if keep_snapshot else "snap-fake-id-01",
+    )
+    source = "pub:https://pub.example.com?task_id=1234"
+
+    requests_mocker.register_uri(
```

Comment:
Personally I don't think this is the right way to set up the test data here.  The test has been tied to a specific implementation detail of how pushsource library obtains data for 'pub' backend when it was unnecessary to do so.  If that implementation detail changes then the test will start to fail.  It's not the job of pubtools-ami's test suite to test the pushsource pub backend.

If the pubtools-ami code can work with any pushsource backend, then I think the better approaches here are:

- use 'staged' backend instead (if easy to do so)
- or just code your own backend - doing that in a test is an intended usage.

For the latter I mean similarly to how some tests in pubtools-pulp do it today:

```python
    fake_items = [
        FilePushItem(name="file1", dest=["some-repo", "other-repo", "/some/path"]),
        FilePushItem(name="file2", dest=["/some/path", "/other/path"]),
        FilePushItem(name="file3", dest=["final-repo"]),
    ]
    Source.register_backend("fake", lambda: fake_items)

    # ... then later pass 'fake:' URI to the code under test
```

One of the reasons pushsource has an extensible API is so that you can write tests like the above which simply hardcode whatever push items you want and avoid a dependency on the implementation details of any specific backend.

Reply:
good suggestion, I'll rewrite that test data setup.